proposals/0000-swift-settings.md

```diff
@@ -0,0 +1,446 @@
+possible if defaultIsolation was defined in the standard library since `Actor`
+is defined in Concurrency,
+
+Each static method on `SwiftSetting` is expecting to return a `SwiftSetting`
+struct. The actual value returned is not important since this code will not be
+executed.
+
+`#SwiftSetting` will only support an explicit opted-in group of command line
+flags. This initially will only include default isolation. All such options must
+possess the following characteristics to be compatible with `#SwiftSettings`:
+
+* No impact on parsing
+
+* Only modify the program in a manner that can be reflected in a textual interface
+  without the need for `#SwiftSettings` to be serialized into a textual
+  interface file.
+
+* Does not cause module wide effects that cannot be constrained to a single
+  file. An example of such a setting is `enable-library-evolution`.
+
+In the future, more options can be added as appropriate (see future
+directions). In order to ensure that any such options also obey the above
+restrictions, we expect that any feature that wants to be supported by
+`#SwiftSettings` must as part of their swift-evolution process consider the ABI
+and source stability impacts of adding support to `#SwiftSettings`. At minimum,
+the feature must obey the above requirements.
+
+Each argument can be passed to `#SwiftSettings` exactly once. Otherwise, an
+error will be emitted:
+
+```swift
+#SwiftSetting(
+  .defaultIsolation(MainActor.self),
+  .defaultIsolation(nil) // Error!
+)
+
+#SwiftSetting(
+  .defaultIsolation(MainActor.self) // Error!
+)
+```
+
+If a command line option is set to different values at the module and file
+level, the setting at the file level will take precedence.
+
+## Source compatibility
+
+The addition of the `#SwiftSettings` and `SwiftSetting` cannot impact the
+parsing of other constructs since the name shadowing rules in the compiler will
+ensure that any local macros or types that shadow those names will take
+precedence. If the user wishes to still use `#SwiftSettings`, the user can spell
+the macro as `#Swift.SwiftSettings` as needed.
+
+Importantly this means that `#SwiftSettings` if used in the swiftpm code base or
+in `Package.swift` files would always need to be namespaced. The authors think
+that this is an acceptable trade-off since:
+
+* The `swiftpm` codebase is one project out of many and can just namespace as
+  appropriate.
+* `Package.swift` files really shouldn't need `#SwiftSettings` since they are
+  very self contained APIs.
+
+## ABI compatibility
+
+This proposal does not inherently break ABI. But it can break ABI if a command
+line option is enabled that causes the generated code's ABI to change. As part
+of enabling an option, one must consider such implications.
+
+## Implications on adoption
+
+Adopters of this proposal should be aware that while this feature does not
+inherently break ABI, enabling options using `#SwiftSettings` that break ABI can
+result in ABI breakage.
```

Comment:
You can be more explicit here - say that changing default actor isolation on public API is not source compatible and not ABI compatible.